src/components/Login/index.tsx

```diff
@@ -1,34 +1,40 @@
-import { USERS } from 'fakeData/users';
+import CircleLoading from 'components/Loading/CircleLoading';
 import React from 'react';
-import { useHistory } from 'react-router-dom';
-import { login } from 'utils/auth';
+import { Redirect } from 'react-router-dom';
 import './Login.scss';
+import { useLogin } from './useLogin';
 import { useLoginForm } from './useLoginForm';
 
 function Login() {
-   const history = useHistory();
-
+   const { login, loading, error } = useLogin();
    // handle login
-   const handleLogin = () => {
+   const handleLogin = async () => {
       const { username, password } = values;
       if (username && password) {
-         const user = USERS.find((user) => {
-            return user.username === username && user.password === password;
-         });
-
-         if (!user) {
-            alert('Login information is incorrect!');
-         } else {
-            login(user);
-            resetForm();
-            history.push('/');
-            window.location.reload();
-         }
+         await login(username, password);
+         resetForm();
       }
    };
+
    const { values, errors, handleChange, handleSubmit, resetForm } =
       useLoginForm(handleLogin);
 
+   if (error) {
+      return (
+         <Redirect
+            push
+            to={{
+               pathname: '/error',
+               state: { error: error },
+            }}
+         />
+      );
+   }
```

Comment:
E chưa sửa chỗ này. Nếu error thì show 1 cái alert message dưới `Login to Dashboard` header (Thông tin đăng nhập k đúng)

![image](https://user-images.githubusercontent.com/31428578/138263136-7f615a15-f557-47f4-ab67-ee6b3def1f72.png)